### PR TITLE
NETOBSERV-1231: Improve promencode API: scaling & filters

### PR DESCRIPTION
**Pull request by jotak**
- New ValueScale config to allow scale conversion, e.g. RTT ns to seconds
- To avoid having reserved characters or words in filters, use a dedicated API instead: new setting Filter.Type (exact,presence,absence or regex)
- Pre-compute regex by setting up a "predicate" function per filter

**Breaking-change** : metrics currently defining filters such as "FlowDirection: 0|2" must be adapted with the new Filter Type API

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [ ] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [x] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Comment by jotak** [comment]
For info, I set no-qe as this has no visible change; however this PR will help us define better RTT metrics for NETOBSERV-1231

File: pkg/pipeline/encode/encode_prom_test.go [hunk 1]
```diff
@@ -485,8 +524,8 @@ func buildFlow() config.GenericMap {
 }
 
 func hundredFlows() []config.GenericMap {
-	flows := make([]config.GenericMap, 100)
-	for i := 0; i < 100; i++ {
+	flows := make([]config.GenericMap, 1000)
+	for i := 0; i < 1000; i++ {
```
Comment (jpinsonneau) [inline pkg/pipeline/encode/encode_prom_test.go:528]: rename the function above then :nerd_face:

**Review by jpinsonneau — COMMENTED** [review]
Sounds good, just a small comment in the code.  You also need to run `make docs` to update `api.md` file.

On top of that, can you please either put [NETOBSERV-1231](https://issues.redhat.com//browse/NETOBSERV-1231) in the title or create another Jira task for this ?

Thanks @jotak !

File: pkg/pipeline/encode/encode_prom_test.go (hunk 1 above)
Comment (jotak) [reply pkg/pipeline/encode/encode_prom_test.go:528]: done!

**Review by OlivierCazade — APPROVED** [review]

**Pull request merged**